src/pip/_internal/operations/prepare.py

```diff
@@ -131,10 +129,12 @@ def get_http_url(
         from_path = already_downloaded_path
         content_type = mimetypes.guess_type(from_path)[0]
     else:
-        # let's download to a tmp dir
-        from_path, content_type = _download_http_url(
-            link, downloader, temp_dir.path, hashes
-        )
+        # TODO: Download directly to download_dir instead of
```

Comment:
The comment is referring to the following snippet

https://github.com/pypa/pip/blob/4c7bbdb4e42dffd37c881105c94512b8836731fc/src/pip/_internal/operations/prepare.py#L545-L555

which IMHO can (or should?) be avoided, especially the copying part.  If it's not a problem, let's leave the comment here to be addressed later once I'm done messing with the relevant routines in the next few PRs :smile:

Reply:
A TODO comment here is, IMO, a distraction from the more pressing work we're trying to do. For this particular issue I think an extra copy is worth the relatively clearer separation of concerns that we have now. I would really prefer that we break it into an issue, since that tends to foster better discussion compared to PRs, can be prioritized, and doesn't risk the burden passing to someone else.

Reply:
I removed the TODO and made a note to my personal TODO to convert this TODO to an issue.